pkg/manager/controllers/addonconfiguration/addon_configuration_reconciler.go

```diff
@@ -72,15 +75,50 @@ func (d *managedClusterAddonConfigurationReconciler) reconcile(
 		errs = append(errs, err)
 	}
 
-	for _, addon := range graph.addonToUpdate() {
+	nonplacementaddons, placementaddons := graph.addonToUpdateByPlacement()
+	// update all non placement addons
+	if err := d.updateAddonsConfig(ctx, nonplacementaddons); err != nil {
+		errs = append(errs, err...)
+	}
+
+	// update placement addons
+	for _, placementStrategy := range cma.Spec.InstallStrategy.Placements {
+		rollouttype := placementStrategy.RolloutStrategy.Type
```

Comment:
should we just put RolloutStrategy into the graph, so we do not check cma here again

Reply:
fixed